public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -163,6 +163,30 @@ export class LokiDatasource
       ...options.scopedVars,
       ...this.getRangeScopedVars(options.range),
     };
+
+    // if all these are true, run query through backend:
+    // - feature-flag is enabled
+    // - we are in explore-mode
+    // - for every query it is true that:
+    //   - query is range query
+    //   - and query is metric query
+    //   - and query is not a log-volume-query (those need a custom http header)
+    const shouldRunBackendQuery =
+      config.featureToggles.lokiBackendMode &&
+      options.app === CoreApp.Explore &&
+      options.targets.every(
+        (query) => query.queryType === LokiQueryType.Range && isMetricsQuery(query.expr) && !query.volumeQuery
```

Comment:
If I run two queries, one range and one instant - they both go through the proxy. Even if I turn off the instant query, they still both go through the proxy. Maybe we should only pass enabled queries into `options.targets.every`?

Reply:
That sounds fair enough :)